src/provider/provider_level_zero.c

```diff
@@ -553,6 +481,133 @@ static umf_result_t ze_memory_provider_free(void *provider, void *ptr,
         g_ze_ops.zeMemFreeExt(ze_provider->context, &desc, ptr));
 }
 
+static umf_result_t query_min_page_size(ze_memory_provider_t *ze_provider,
+                                        size_t *min_page_size) {
+    assert(min_page_size);
+
+    LOG_DEBUG("Querying minimum page size");
+
+    void *ptr;
+    umf_result_t result = ze_memory_provider_alloc(ze_provider, 1, 0, &ptr);
+    if (result != UMF_RESULT_SUCCESS) {
+        return result;
+    }
+
+    ze_memory_allocation_properties_t properties = {
+        .stype = ZE_STRUCTURE_TYPE_MEMORY_ALLOCATION_PROPERTIES};
+    ze_result_t ze_result = g_ze_ops.zeMemGetAllocProperties(
```

Comment:
Have you considered `zeVirtualMemQueryPageSize` instead of your current implementation?

Reply:
As far as I know that function is used for zeVirtualMemReserve and zePhysicalMemCreate only. It can theoretically be different than what is used by regular allocations.